automem/utils/text.py

```
@@ -96,3 +99,105 @@ def _extract_keywords(text: str) -> List[str]:
+                len(content),
+                len(summary),
+                (1 - len(summary) / len(content)) * 100,
+            )
+            return summary
+
+        logger.warning(
+            "Summarization did not reduce content length (%d -> %d), returning original",
+            len(content),
+            len(summary) if summary else 0,
+        )
+        return None
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)